camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/catalog/CatalogYamlRebindTest.java

```diff
@@ -85,63 +95,97 @@
-    public void testRebindWithCatalogPropertiesForDeprecationAndEnablementAbsent() throws Exception {
-        runRebindWithCatalogAndApp(RebindWithCatalogTestMode.STRIP_DEPRECATION_AND_ENABLEMENT_FROM_CATALOG_ITEM);
+
+    @DataProvider
+    public Object[][] dataProvider() {
+        return new Object[][] {
+            {RebindWithCatalogTestMode.NO_OP, false},
+            {RebindWithCatalogTestMode.NO_OP, true},
+            
+            {RebindWithCatalogTestMode.STRIP_DEPRECATION_AND_ENABLEMENT_FROM_CATALOG_ITEM, false},
+            {RebindWithCatalogTestMode.STRIP_DEPRECATION_AND_ENABLEMENT_FROM_CATALOG_ITEM, true},
+            
+            {RebindWithCatalogTestMode.DEPRECATE_CATALOG, false},
+            {RebindWithCatalogTestMode.DEPRECATE_CATALOG, true},
+            
+            {RebindWithCatalogTestMode.DISABLE_CATALOG, false},
+            {RebindWithCatalogTestMode.DISABLE_CATALOG, true},
+            
+            // For DELETE_CATALOG, see https://issues.apache.org/jira/browse/BROOKLYN-149.
+            // Deletes the catalog item before rebind, but the referenced types are still on the 
+            // default classpath. Will fallback to loading from classpath.
+            //
+            // Does not work for OSGi, because our bundle will no longer be available.
+            {RebindWithCatalogTestMode.DELETE_CATALOG, false},
+            
+            // Upgrades the catalog item before rebind, deleting the old version.
+            // Will automatically upgrade. Test will enable "FEATURE_AUTO_FIX_CATALOG_REF_ON_REBIND"
+            {RebindWithCatalogTestMode.REPLACE_CATALOG_WITH_NEWER_VERSION, false},
+            {RebindWithCatalogTestMode.REPLACE_CATALOG_WITH_NEWER_VERSION, true},
+        };
     }
 
-    @SuppressWarnings({ "unused", "deprecation" })
-    protected void runRebindWithCatalogAndApp(RebindWithCatalogTestMode mode) throws Exception {
+    @Test(dataProvider = "dataProvider")
+    @SuppressWarnings("deprecation")
+    public void testRebindWithCatalogAndApp(RebindWithCatalogTestMode mode, boolean useOsgi) throws Exception {
+        if (mode == RebindWithCatalogTestMode.REPLACE_CATALOG_WITH_NEWER_VERSION) {
+            BrooklynFeatureEnablement.enable(BrooklynFeatureEnablement.FEATURE_AUTO_FIX_CATALOG_REF_ON_REBIND);
+        }
+
         String appSymbolicName = "my.catalog.app.id.load";
         String appVersion = "0.1.0";
-        String appCatalogFormat = Joiner.on("\n").join(
-                "brooklyn.catalog:",
-                "  id: " + appSymbolicName,
-                "  version: %s",
-                "  itemType: entity",
-                "  item:",
-                "    type: "+ BasicEntity.class.getName(),
-                "    brooklyn.enrichers:",
-                "    - type: "+TestEnricher.class.getName(),
-                "    brooklyn.policies:",
-                "    - type: "+TestPolicy.class.getName());
         
+        String appCatalogFormat;
+        if (useOsgi) {
```

Comment:
Add a `TestResourceUnavailableException.throwIfResourceUnavailable(getClass(), OSGI_BUNDLE_URL);` for this case